### PR TITLE
put unsigged defaults in the tree

### PR DESCRIPTION
**Pull request by ptarjan**
As suggested by @elliottt in https://github.com/sorbet/sorbet/pull/1994#discussion_r337627230 we now typecheck this case. Yay.

### Test plan

See included automated tests.

### CODE REVIEW

File: resolver/resolver.cc [hunk 1]
```diff
@@ -1455,7 +1455,9 @@ class ResolveSignaturesWalk {
 
             if (auto *optArgExp = ast::cast_tree<ast::OptionalArg>(argExp.get())) {
                 if (!argType) {
-                    optArgExp->default_ = nullptr;
+                    auto assign =
+                        ast::MK::Assign(optArgExp->loc, optArgExp->expr->deepCopy(), move(optArgExp->default_));
```
Comment (dmitry-stripe) [inline resolver/resolver.cc]: why do we deep copy `optArgExp->expr`.
Where does original go?

Comment (ptarjan) [reply resolver/resolver.cc]: @dmitry-stripe it is used by the cfg. Without the deep copy i get:

```
[2019-10-22 21:52:30.447] [fatalFallback] [error] Exception::raise(): nullptr passed to typecase

[2019-10-22 21:52:31.038] [fatalFallback] [error] Backtrace:
sorbet::ast::MK::arg2Local(sorbet::ast::Expression*) (in liblsp.so) + 116
sorbet::cfg::CFGBuilder::buildFor(sorbet::core::Context, sorbet::ast::MethodDef&) (in libbuilder.so) + 2248
sorbet::test::CFGCollectorAndTyper::preTransformMethodDef(sorbet::core::Context, std::__1::unique_ptr<sorbet::ast::MethodDef, std::__1::default_delete<sorbet::ast::MethodDef> >) (in test_corpus_runner) (test_corpus.cc:108)
sorbet::ast::PostPonePreTransform_MethodDef<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext, true>::call(sorbet::core::MutableContext, std::__1::unique_ptr<sorbet::ast::MethodDef, std::__1::default_delete<sorbet::ast::MethodDef> >, sorbet::test::CFGCollectorAndTyper&) (in test_corpus_runner) (treemap.h:204)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapMethodDef(std::__1::unique_ptr<sorbet::ast::MethodDef, std::__1::default_delete<sorbet::ast::MethodDef> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:293)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapIt(std::__1::unique_ptr<sorbet::ast::Expression, std::__1::default_delete<sorbet::ast::Expression> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:666)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapClassDef(std::__1::unique_ptr<sorbet::ast::ClassDef, std::__1::default_delete<sorbet::ast::ClassDef> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:281)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapIt(std::__1::unique_ptr<sorbet::ast::Expression, std::__1::default_delete<sorbet::ast::Expression> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:672)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapInsSeq(std::__1::unique_ptr<sorbet::ast::InsSeq, std::__1::default_delete<sorbet::ast::InsSeq> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:598)
sorbet::ast::TreeMapper<sorbet::test::CFGCollectorAndTyper, sorbet::core::MutableContext>::mapIt(std::__1::unique_ptr<sorbet::ast::Expression, std::__1::default_delete<sorbet::ast::Expression> >, sorbet::core::MutableContext) (in test_corpus_runner) (treemap.h:668)
std::__1::unique_ptr<sorbet::ast::Expression, std::__1::default_delete<sorbet::ast::Expression> > sorbet::ast::TreeMap::apply<sorbet::core::MutableContext, sorbet::test::CFGCollectorAndTyper>(sorbet::core::MutableContext, sorbet::test::CFGCollectorAndTyper&, std::__1::unique_ptr<sorbet::ast::Expression, std::__1::default_delete<sorbet::ast::Expression> >) (in test_corpus_runner) (treemap.h:712)
sorbet::test::ExpectationTest_PerPhaseTest_Test::TestBody() (in test_corpus_runner) (test_corpus.cc:414)
void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in libgtest.so) (gtest.cc:2439)
void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in libgtest.so) (gtest.cc:2475)
testing::Test::Run() (in libgtest.so) (gtest.cc:2521)
testing::TestInfo::Run() (in libgtest.so) (gtest.cc:2693)
testing::TestSuite::Run() (in libgtest.so) (gtest.cc:2821)
testing::internal::UnitTestImpl::RunAllTests() (in libgtest.so) (gtest.cc:5341)
bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in libgtest.so) (gtest.cc:2439)
bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in libgtest.so) (gtest.cc:2475)
testing::UnitTest::Run() (in libgtest.so) (gtest.cc:4930)
RUN_ALL_TESTS() (in test_corpus_runner) (gtest.h:2472)
main (in test_corpus_runner) (test_corpus.cc:1252)
start (in libdyld.dylib) + 1
```

Comment (ptarjan) [reply resolver/resolver.cc]: Should that use the symbol instead of the ast?

Comment (DarkDimius) [reply resolver/resolver.cc]: It seems that the change left `nullptr` in the tree, which makes for ill-constructed tree.

**Pull request merged**